### PR TITLE
clean up dependencies and a few other fixes

### PR DESCRIPTION
**Pull request by rkuhn**
- update patch versions on Netty, Scalatest, scala-xml
- remove inter-project dependencies between
  streamTestkit -> peristence -> remote
- make PartitionOutOfBoundsException extend IndexOutOfBoundsException
- some weird formatting
- remove -experimental from HTTP Testkit/Tests/Marshallers

### CODE REVIEW

File: project/AkkaBuild.scala [hunk 1]
```diff
@@ -181,7 +181,7 @@ object AkkaBuild extends Build {
   lazy val persistence = Project(
     id = "akka-persistence",
     base = file("akka-persistence"),
-    dependencies = Seq(actor, remote % "test->test", testkit % "test->test", protobuf)
```
Comment (drewhk) [inline project/AkkaBuild.scala]: lol

Comment (drewhk) [inline project/AkkaBuild.scala]: was it for the test "distributed journal" (the remoting+LevelDB based one)?

**Comment by drewhk** [comment]
LGTM

**Comment by rkuhn** [comment]
Hmm, IncompatibleClassChangeError. Will leave this until Monday to be sorted out with the help of the Jenkins experts.

**Comment by rkuhn** [comment]
fixes #19755

File: akka-http-marshallers-scala/akka-http-spray-json/build.sbt [hunk 2]
```diff
@@ -2,7 +2,6 @@ import akka._
 import com.typesafe.tools.mima.plugin.MimaKeys
 
 AkkaBuild.defaultSettings
-AkkaBuild.experimentalSettings
```
Comment (ktoso) [inline akka-http-marshallers-scala/akka-http-spray-json/build.sbt]: Hm, perhaps OK to make stable now... Was initially thinking to un-experimental it together with DSL and JSON "streaming". I guess nothing breaking will happen here though, just additions.

File: project/AkkaBuild.scala (hunk 1 above)
Comment (ktoso) [inline project/AkkaBuild.scala]: hm no idea, but certainly not needed now :)

**Comment by ktoso** [comment]
LGTM

**Comment by rkuhn** [comment]
Okay, it still fails with these IncompatibleClassChangeErrors, I give up. Who can figure this out?

**Comment by drewhk** [comment]
wat, and only *some* of the tests fail on Jenkins. In fact, it looks rather random which ones do (every subproject have a few failing ones, seemingly unrelated)

**Comment by rkuhn** [comment]
This can only mean that multiple jobs are running in parallel and overwriting each others’ class files.

**Comment by ktoso** [comment]
I started reviewing the new Java DSL and noticed it's touching the marshallers and also have other worries on making the marshaller stable tomorrow. Can we keep them experimental until 2.4.3 please?

File: akka-http-core/src/test/scala/akka/http/javadsl/HttpExtensionApiSpec.scala [hunk 3]
```diff
@@ -13,7 +13,7 @@ import akka.http.javadsl.model.ws._
 import akka.http.javadsl.settings.{ ClientConnectionSettings, ConnectionPoolSettings, ServerSettings }
 import akka.japi.Pair
```
Comment (vasseurjb) [inline akka-http-core/src/test/scala/akka/http/javadsl/HttpExtensionApiSpec.scala]: Codacy Issue found: [Use brackets to group imports from the same package](https://www.codacy.com/app/vasseurjb/akka/file/2046322363/issues/source?bid=3132306&fileBranchId=3132371#l14)

File: akka-stream/src/main/scala/akka/stream/javadsl/Source.scala [hunk 4]
```diff
@@ -777,38 +777,37 @@ final class Source[+Out, +Mat](delegate: scaladsl.Source[Out, Mat]) extends Grap
...
+   * '''Backpressures when''' downstream backpressures or there are still remaining elements from the
+   * previously calculated collection
+   *
+   * '''Completes when''' upstream completes and all remaining elements has been emitted
+   *
+   * '''Cancels when''' downstream cancels
+   */
   def statefulMapConcat[T](f: function.Creator[function.Function[Out, java.lang.Iterable[T]]]): javadsl.Source[T, Mat] =
-    new Source(delegate.statefulMapConcat{ () ⇒ {
+    new Source(delegate.statefulMapConcat { () ⇒
```
Comment (vasseurjb) [inline akka-stream/src/main/scala/akka/stream/javadsl/Source.scala]: Codacy Issue found: [Non ascii characters are not allowed](https://www.codacy.com/app/vasseurjb/akka/file/2046323167/issues/source?bid=3132306&fileBranchId=3132371#l807)

**Comment by rkuhn** [comment]
Merging manually (test failure was VirtualPublisher TCK).

**Pull request merged**